docs/config-file/v2.rst

```diff
@@ -0,0 +1,365 @@
+   See :ref:`config-file/v2:python.install`.
+
+Behind the scene the following Pip command will be run::
+
+   $ pip install -e .[tests,docs]
+
+python.system_packages
+``````````````````````
+
+Give the virtual environment access to the global site-packages directory.
+
+:Type: ``bool``
+:Default: ``false``
+
+conda
+~~~~~
+
+Configuration for Conda support.
+
+.. code-block:: yaml
+
+   conda:
+     environment: environment.yml
+
+conda.environment
+`````````````````
+
+The path to the Conda environment file from the root of the project.
+
+:Type: ``path``
+:Required: ``true``
+
+build
+~~~~~
+
+Configuration for the documentation build process.
+
+.. code-block:: yaml
+
+   build:
+     image: latest
+
+   python:
+     version: 3.6
+
+
+build.image
+```````````
+
+The build Docker image to be used.
+
+:Type: ``string``
+:Options: ``stable``, ``latest``
+
+Each image support different Python versions and has different packages installed,
+as defined here:
+
+* `stable <https://github.com/rtfd/readthedocs-docker-images/tree/releases/2.x>`_: :supportedversions:`2.0`
+* `latest <https://github.com/rtfd/readthedocs-docker-images/tree/releases/3.x>`_: :supportedversions:`3.0`
+
+sphinx
+~~~~~~
+
+Configuration for Sphinx documentation
+(this is the default documentation type).
+
+.. code-block:: yaml
+
+   sphinx:
+     builder: html
+     configuration: conf.py
+     fail_on_warning: true
+
+sphinx.builder
+``````````````
+
+The builder type for the Sphinx documentation.
+
+:Type: ``string``
+:Options: ``html``, ``htmldir``, ``singlehtml``
+:Default: ``html``
+
+sphinx.configuration
+````````````````````
+
+The path to the ``conf.py`` file.
+
+:Type: ``path``
+:Default: ``null``
+
+If the value is ``null``,
+Read the Docs would try to find a ``conf.py`` file in your project.
+
+sphinx.fail_on_warning
+``````````````````````
+
+Turn warnings into errors.
+This means that the build stops at the first warning and exits with exit status 1.
+
+:Type: ``bool``
+:Default: ``false``
+
+mkdocs
+~~~~~~
+
+Configuration for Mkdocs documentation.
+
+.. code-block:: yaml
+
+   mkdocs:
+     configuration: mkdocs.yml
+     fail_on_warning: false
+
+mkdocs.configuration
+````````````````````
+
+The path to the ``mkdocs.yml`` file
+
+:Type: ``path``
+:Default: ``null``
+
+If the value is ``null``,
+Read the Docs would try to find a ``mkdocs.yml`` file in your project.
+
+mkdocs.fail_on_warning
+``````````````````````
+
+Turn warnings into errors.
+This means that the build stops at the first warning and exits with exit status 1.
+
+:Type: ``bool``
+:Default: ``false``
+
+submodules
+~~~~~~~~~~
+
+Git submodules configuration.
+
+.. note::
+   
+   You can't include and exclude submodules at the same time.
+
+.. code-block:: yaml
+   
+   submodules:
+     include: all
+     recursive: true
+
+submodules.include
+``````````````````
+
+List of submodules to be included.
+
+:Type: ``list``
+:Default: ``[]``
```

Comment:
I found this a little confusing.

We are saying that `[]` is the default. What this means when `recursive: true`? I'd say that _none_ of the submodules will be cloned, but I suppose that's not what we are doing.

Reply:
When is recursive: true and [], we don't sync the submodules, recursive is ignored

Reply:
Maybe it worth to have a note about this here since this behavior is tricky.